gooddata-sdk/gooddata_sdk/catalog/workspace/declarative_model/workspace/logical_model/ldm.py

```diff
@@ -61,14 +120,15 @@ def store_to_disk(self, workspace_folder: Path) -> None:
     Args:
         data_source_mapping (dict):
             Key value pairs representing which DS(key) should be replaced by which DS(value).
+            If mapping is empty, noop
 
     Returns:
         self
     """
 
-    def modify_mapped_data_source(self, data_source_mapping: dict) -> CatalogDeclarativeModel:
-        if self.ldm is not None:
-            for dataset in self.ldm.datasets:
+    def modify_mapped_data_source(self, data_source_mapping: Optional[dict]) -> CatalogDeclarativeLdm:
```

Comment:
Why is `data_source_mapping` now optional? Is it because of chaining? If yes we can pass the empty dictionary and the behavior should be the same and we will preserve data_source_mapping to be mandatory.

Reply:
Yes, it supports chaining.
If you chain multiple methods and one of their input is optional from an outside scope, it would force developers to implement nasty IF statements.
Extended DOC string to explain it.